src/pyxplor/plot_numeric.py

```diff
@@ -49,13 +52,68 @@ def plot_numeric(
 
     Returns
     -------
-    None
-
+    tuple
+        A tuple containing the `fig` and `ax` objects.
 
     Examples
     --------
     numeric = ["numeric_var1", "numeric_var2"]
-    plot_numeric(df, numeric, "hist+kde", 30, 10, (15, 20))
+    plot_numeric(df, numeric, "hist+kde", 30, 10, (10, 15))
     """
+    # Get numeric columns from the dataframe
+    numeric_columns = input_df.select_dtypes(include=['number']).columns.tolist()
+
+    # Filter numeric columns based on the provided list_of_variables
+    numeric_columns = [var for var in list_of_variables if var in numeric_columns]
+
+# Check if there are any numeric columns to plot
+if not numeric_columns:
+    print("No valid numeric columns found in the provided list_of_variables.")
+    return
+
+# Calculate the number of rows and columns dynamically
+total_plots = len(numeric_columns)
+rows = math.ceil(math.sqrt(total_plots))
+cols = math.ceil(total_plots / rows)
+
+# Create subplots in a grid
+fig, ax = plt.subplots(rows, cols, figsize=figsize)
+ax = ax.flatten()
+
+# Loop through numeric columns and plot
+for i, variable in enumerate(numeric_columns):
+    # Plot histogram or kernel density estimate plot for each variable
+    if plot_kind == 'hist':
+        sns.histplot(x=variable, data=input_df, ax=ax[i], bins=20)
+    elif plot_kind == 'kde':
+        sns.kdeplot(x=variable, data=input_df, ax=ax[i])
+    elif plot_kind == 'hist+kde':
+        sns.histplot(x=variable, data=input_df, ax=ax[i], bins=20, kde=True)
+
+        # Add central tendency labels with different colors
+        mean = input_df[variable].mean()
+        median = input_df[variable].median()
+        mean_label = f'Mean: {mean:.2e}'
+        median_label = f'Median: {median:.2e}'
+        ax[i].axvline(mean, color='orange', linestyle='dashed', linewidth=2, label=mean_label)
+        ax[i].axvline(median, color='red', linestyle='dashed', linewidth=2, label=median_label)
+
+        # Add legend in the top right corner
+        ax[i].legend(loc='upper right')
+
+    ax[i].set_title(variable)  # Set subplot title
+
+# Add super title
+fig.suptitle(super_title, fontsize=super_title_font)
+
+# Adjust layout and save the figure
+plt.tight_layout(rect=[0, 0, 1, 0.96])  # Adjust rect to make space for super title
+plt.savefig(output_path)
```

Comment:
As discussed in lab, (see @phchen5's PR https://github.com/UBC-MDS/PyXplor/pull/41) that we should add a boolean parameter `output` (rather than `output_path`) and adjust saving accordingly:

```suggestion
    if output:
        plt.savefig("numerical_variables.png")
```